src/PAModel/MsAppTest.cs

```diff
@@ -269,56 +274,82 @@ public static void DebugChecksum(string pathToZip, TextWriter log, Dictionary<st
-    
-                                    // Paths to current diff files
-                                    string aPath = normFormDir + "\\" + Path.ChangeExtension(e.Name, null) + "-A.json";
-                                    string bPath = normFormDir + "\\" + Path.ChangeExtension(e.Name, null) + "-B.json";
-
-                                    File.WriteAllBytes(aPath, otherContents);
-                                    File.WriteAllBytes(bPath, key);
-
-                                    // For debugging. Help find exactly where the difference is. 
-                                    for (int i = 0; i < otherContents.Length; i++)
-                                    {
-                                        if (i >= key.Length)
-                                        {
-                                            break;
-                                        }
-                                        if (otherContents[i] != key[i])
-                                        {
-
-                                        }
-                                    }
-                                }
-                                else
-                                {
-                                    // success
+                                    checkPropertyMismatch(entry, otherContents, key, errorContainer);
+#if DEBUG
+                                    debugMismatch(entry, otherContents, key, normFormDir);
+#endif
                                 }
-                                comp.Remove(e.FullName);
+
+                                comp.Remove(entry.FullName);
                             }
                             else
                             {
                                 // Missing file!
-                                Console.WriteLine("FAIL: 2nd has added file: " + e.FullName);
+                                Console.WriteLine("FAIL: 2nd has added file: " + entry.FullName);
                             }
                         }
                     }
                 }
             }
+
+        }
+
+        public static void checkPropertyMismatch(ZipArchiveEntry entry, byte[] otherContents, byte[] key, ErrorContainer errorContainer){
+            
+            // Parse each byte array of the different files
+            JsonElement json1 = JsonDocument.Parse(key).RootElement;
+            JsonElement json2 = JsonDocument.Parse(otherContents).RootElement;
+
+            // Check each property and value in json1 to see if each exists and is equal to json2
+            foreach (var currentProperty1 in json1.EnumerateObject())
+            {
+                // If current property from first json file also exists in the second file
+                if(json2.TryGetProperty(currentProperty1.Name, out JsonElement value2)){
+
+                    // If current property value from first json file is not the same as in second
+                    if(!currentProperty1.Value.Equals(value2)){
```

Comment:
I noticed that as well. You may need to use "Deep Equal" to check for the subkey, but this will only output that "Key2" has changed. You may need to traverse into JObject types to identify that "SubKey1" has changed.

Consider using the [JsonTextReader](https://www.newtonsoft.com/json/help/html/t_newtonsoft_json_jsontextreader.htm) which will enumerate all the values in a file. From here you can use the [Path](https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonReader_Path.htm) property to output the JSON path that might differ (i.e `Key2/SubKey1`)

Reply:
attempting recursive function to iterate through

Reply:
I believe I've resolved this now. Can you take a look? @tomjschneider @jt000

Reply:
Let's chat about this a bit more in standup, I'm not sure my understanding of what we want to report on is entirely in sync.

Reply:
removed recursive